### PR TITLE
Add ORFS-inspired Diagram to Readme; use DetSys Nix Installer

### PR DESCRIPTION
**Pull request by donn**
## Documentation
* Installation documents now use the less-brittle Determinate Systems Nix installer, as well as adding warnings about the `apt` version of Nix.
* Added an OpenROAD Flow Scripts-inspired Diagram to the Readme.

## Testing
* CI now uses Determinate Systems Nix Installer for all Nix installations as well as the Magic Nix Cache Action instead of the nonfunctional attempt at local file-based substituters

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -32,23 +32,52 @@ started. You can discuss OpenLane 2 in the
 channel of the
 [Efabless Open Source Silicon Slack](https://invite.skywater.tools).
 
-```python
```
Comment (kareefardi) [inline Readme.md:35]: was this removal intentional

Comment (donn) [reply Readme.md:35]: yes
